src/BSON/Regex.c

```diff
@@ -206,6 +207,24 @@ PHP_METHOD(Regex, __toString)
 }
 /* }}} */
 
+/* {{{ proto array Regex::jsonSerialize()
+*/
+PHP_METHOD(Regex, jsonSerialize)
+{
+	php_phongo_regex_t *intern;
+
+	if (zend_parse_parameters_none() == FAILURE) {
+		return;
+	}
+
+	intern = Z_REGEX_OBJ_P(getThis());
+
+	array_init_size(return_value, 2);
+	ADD_ASSOC_STRINGL(return_value, "$regex", intern->pattern, intern->pattern_len);
+	ADD_ASSOC_STRINGL(return_value, "$options", intern->flags, intern->flags_len);
```

Comment:
I left a comment in [DRIVERS-331](https://jira.mongodb.org/browse/DRIVERS-331?focusedCommentId=1428306&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-1428306) asking whether we should alphabetically order the regex flag characters (as should be done for raw BSON to support binary comparisons). The implementation here leaves the flags as-is.

If we do start alphabetically ordering the characters in the future, we'll likely do so when the Regex object is initialized and this code shouldn't need to change.